code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -546,7 +546,7 @@ func (sc *StorageSmartContract) getAllocationBlobbers(alloc *StorageAllocation,
 	}
 
 	blobberCh := make(chan blobberResp, len(alloc.BlobberAllocs))
-	errorCh := make(chan error)
+	errorCh := make(chan error, 1)
```

Comment:
it should not be a leak as on line 567 we'r closing errorCh
but now with setting it as len 1, we're blocking the goroutine.. as soon as we get error in any go func, rest will hang until errorCh is read, which is not gonna happen as it'll be stuck on wg.Wait(). so it's a deadlock

the way it is right now looks fine. creating a buffered channel is issue here.
if we have to create buffered, it's length should be len(alloc.BlobberAllocs)
`errorCh := make(chan error, len(alloc.BlobberAllocs))`

Reply:
The errorCh needs to be a buffered channel, otherwise the wg.Wait() will be blocked if error happens. And yes, the buffer size should be `len(alloc.BlobberAlloc)` in case all failed with error.